lib/node_modules/@stdlib/types/index.d.ts

```diff
@@ -593,6 +593,85 @@ declare module '@stdlib/types/array' {
+	*/
+	type RealFloatingPointArrayMap = { // eslint-disable-line @typescript-eslint/consistent-type-definitions
+		'float64': Float64Array;
+		'float32': Float32Array;
+	};
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `complex128`: double-precision complex floating-point numbers
+	* -   `complex64`: single-precision complex floating-point numbers
+	*/
+	type ComplexArrayMap = { // eslint-disable-line @typescript-eslint/consistent-type-definitions
+		'complex64': Complex64Array;
+		'complex128': Complex128Array;
+	};
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `complex64`: single-precision complex floating-point numbers
+	* -   `int32`: 32-bit two's complement signed integers
+	* -   `uint32`: 32-bit unsigned integers
+	* -   `int16`: 16-bit two's complement signed integers
+	* -   `uint16`: 16-bit unsigned integers
+	* -   `int8`: 8-bit two's complement signed integers
+	* -   `uint8`: 8-bit unsigned integers
+	* -   `uint8c`: 8-bit unsigned integers clamped to `0-255`
+	*/
+	type IntegerArrayMap = { // eslint-disable-line @typescript-eslint/consistent-type-definitions
+		'int32': Int32Array;
+		'uint32': Uint32Array;
+		'int16': Int16Array;
+		'uint16': Uint16Array;
+		'int8': Int8Array;
+		'uint8': Uint8Array;
+		'uint8c': Uint8ClampedArray;
+	};
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `float64`: double-precision floating-point numbers (IEEE 754)
+	* -   `float32`: single-precision floating-point numbers (IEEE 754)
+	* -   `complex128`: double-precision complex floating-point numbers
+	* -   `complex64`: single-precision complex floating-point numbers
+	* -   `int32`: 32-bit two's complement signed integers
+	* -   `uint32`: 32-bit unsigned integers
+	* -   `int16`: 16-bit two's complement signed integers
+	* -   `uint16`: 16-bit unsigned integers
+	* -   `int8`: 8-bit two's complement signed integers
+	* -   `uint8`: 8-bit unsigned integers
+	* -   `uint8c`: 8-bit unsigned integers clamped to `0-255`
+	*/
+	type TypedArrayMap = RealFloatingPointArrayMap & ComplexArrayMap & IntegerArrayMap;
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `float64`: double-precision floating-point numbers (IEEE 754)
+	* -   `float32`: single-precision floating-point numbers (IEEE 754)
+	* -   `complex128`: double-precision complex floating-point numbers
+	* -   `complex64`: single-precision complex floating-point numbers
+	* -   `int32`: 32-bit two's complement signed integers
+	* -   `uint32`: 32-bit unsigned integers
+	* -   `int16`: 16-bit two's complement signed integers
+	* -   `uint16`: 16-bit unsigned integers
+	* -   `int8`: 8-bit two's complement signed integers
+	* -   `uint8`: 8-bit unsigned integers
+	* -   `uint8c`: 8-bit unsigned integers clamped to `0-255`
+	* -   `generic`: generic JavaScript values
+	*/
+	type ArrayDataTypeMap<T> = TypedArrayMap & { 'generic': Array<T> };
```

Comment:
I am not in favor of this sort of naming inconsistency. As a declaration author, I have to remember special cases, which I am not inclined to do.

I'd also say we should just be pedantic and create `*DataTypeMap` declarations for all corresponding data type ["kinds"](https://github.com/stdlib-js/stdlib/blob/43af7eb290aa0262006057f4358cf45e7ba9ec84/lib/node_modules/%40stdlib/types/index.d.ts#L42). Again, I don't want to have to think about which type maps have already been created and which have not. Since we are here, just create all of them.

Reply:
Agreed. Was planning to rename after settling on a naming convention. Adding all the remaining DataTypeMap declarations...